lib/std/x/os/socket_windows.zig

```diff
@@ -0,0 +1,392 @@
+    /// with a set of flags specified. It returns the number of bytes that are
+    /// written to the socket.
+    pub fn writeVectorized(self: Socket, msg: os.msghdr_const, flags: u32) !usize {
+        return error.NotImplemented;
+    }
+
+    /// Read multiple I/O vectors with a prepended message header from the socket
+    /// with a set of flags specified. It returns the number of bytes that were
+    /// read into the buffer provided.
+    pub fn readVectorized(self: Socket, msg: *os.msghdr, flags: u32) !usize {
+        return error.NotImplemented;
+    }
+
+    /// Query the address that the socket is locally bounded to.
+    pub fn getLocalAddress(self: Socket) !Socket.Address {
+        var address: ws2_32.sockaddr_storage = undefined;
+        var address_len: c_int = @sizeOf(ws2_32.sockaddr_storage);
+
+        const rc = ws2_32.getsockname(self.fd, @ptrCast(*ws2_32.sockaddr, &address), &address_len);
+        if (rc == ws2_32.SOCKET_ERROR) {
+            return switch (ws2_32.WSAGetLastError()) {
+                .WSANOTINITIALISED => unreachable,
+                .WSAEFAULT => unreachable,
+                .WSAENETDOWN => error.NetworkSubsystemFailed,
+                .WSAENOTSOCK => error.FileDescriptorNotASocket,
+                .WSAEINVAL => error.SocketNotBound,
+                else => |err| windows.unexpectedWSAError(err),
+            };
+        }
+
+        return Socket.Address.fromNative(@alignCast(4, @ptrCast(*os.sockaddr, &address)));
+    }
+
+    /// Query the address that the socket is connected to.
+    pub fn getRemoteAddress(self: Socket) !Socket.Address {
+        var address: ws2_32.sockaddr_storage = undefined;
+        var address_len: c_int = @sizeOf(ws2_32.sockaddr_storage);
+
+        const rc = ws2_32.getpeername(self.fd, @ptrCast(*ws2_32.sockaddr, &address), &address_len);
+        if (rc == ws2_32.SOCKET_ERROR) {
+            return switch (ws2_32.WSAGetLastError()) {
+                .WSANOTINITIALISED => unreachable,
+                .WSAEFAULT => unreachable,
+                .WSAENETDOWN => error.NetworkSubsystemFailed,
+                .WSAENOTSOCK => error.FileDescriptorNotASocket,
+                .WSAEINVAL => error.SocketNotBound,
+                else => |err| windows.unexpectedWSAError(err),
+            };
+        }
+
+        return Socket.Address.fromNative(@alignCast(4, @ptrCast(*os.sockaddr, &address)));
+    }
+
+    /// Query and return the latest cached error on the socket.
+    pub fn getError(self: Socket) !void {
+        return {};
+    }
+
+    /// Query the read buffer size of the socket.
+    pub fn getReadBufferSize(self: Socket) !u32 {
+        return 0;
+    }
+
+    /// Query the write buffer size of the socket.
+    pub fn getWriteBufferSize(self: Socket) !u32 {
+        return 0;
+    }
+
+    /// Set a socket option.
+    pub fn setOption(self: Socket, level: u32, name: u32, value: []const u8) !void {
```

Comment:
`name` doesn't seem like a good name for this variable?

Reply:
How about `option: u32`?

Reply:
Changed it to `code: u32`.